app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = dict()
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(list_of_people: list) -> list:
+    our_list = []
+
+    for person in list_of_people:
+        person = Person(person["name"], person["age"])
+        our_list.append(person)
+
+    for person in list_of_people:
+        if person.get("wife"):
+            wife_name = person["wife"]
+            wife = Person.people[wife_name]
+            name_of_person = person["name"]
+            person = Person.people[name_of_person]
+            person.wife = wife
```

Comment:
Do not allocate so many variables and avoid code duplication while retrieving `person`. You could use walrus operator to improve code efficiency as well
```suggestion
    for person_data in list_of_people:
        person = Person.people[person_data["name"]
        if wife_name := person_data.get("wife"):
            person.wife = Person.people[wife_name]
```

Apply a similar change to a husband as well